### PR TITLE
[SYCL] Pre-process footer file when -include-footer is used.

### PR DESCRIPTION
**Pull request by zahiraam**
With the new option `-include-footer=file.h`, the name `file.h` and its content need to be pulled in at the end of the pre-processed file. The actual pre-processed file should have this form:

header.h
file.cpp
content of file.cpp
integration-footer.h
content-of-footer.h

### CODE REVIEW

File: clang/test/Preprocessor/Inputs/file2.h.new [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#include "footer2.h"
```
Comment (elizabethandrews) [inline clang/test/Preprocessor/Inputs/file2.h.new:1]: Was this file uploaded by mistake?

File: clang/test/Preprocessor/sycl-int-footer.cpp [hunk 2]
```diff
@@ -1,22 +1,47 @@
...
+// CHECKFOOTER1-NEXT: int file2() {
+// CHECKFOOTER1-NEXT:   return 2;
+// CHECKFOOTER1-NEXT: }
+
+// CHECKFOOTER2: # 1 "[[INPUTFILE_CF2:.+\.cpp]]"
+// CHECKFOOTER2: # 1 "[[INTHEADER:.+\.h]]" 1
+// CHECKFOOTER2-NEXT: # 1 "[[INTHEADER3:.+\.h]]" 1
+// CHECKFOOTER2-NEXT: # 1 "[[INTHEADER4:.+\.h]]" 1
+// CHECKFOOTER2: void foo3();
+// CHECKFOOTER2-NEXT: # 2 "[[INTHEADER3:.+\.h]]" 2
```
Comment (premanandrao) [inline clang/test/Preprocessor/sycl-int-footer.cpp:32]:
```suggestion
// CHECKFOOTER2-NEXT: # 2 "[[INTHEADER3]]" 2
```

File: clang/test/Preprocessor/Inputs/file2.h.new (hunk 1 above)
Comment (zahiraam) [reply clang/test/Preprocessor/Inputs/file2.h.new:1]: This file is included in the `footer.h` file.

Comment (premanandrao) [reply clang/test/Preprocessor/Inputs/file2.h.new:1]: Are you talking about ``footer2.h``?  I think @elizabethandrews is asking about ``file2.h.new``.

Comment (zahiraam) [reply clang/test/Preprocessor/Inputs/file2.h.new]: Removed.

**Review by premanandrao — APPROVED** [review]
I am okay with the changes.   I presume this has been tested along with @mdtoguchi's Driver changes offline.

**Comment by zahiraam** [comment]
> I am okay with the changes. I presume this has been tested along with @mdtoguchi's Driver changes offline.

We should wait before proceeding with this. Still working on the other issue found when not in pre-processing mode. A few things might change here when that other issued is fixed.

**Pull request closed without merging**